scripts/joysetup.sh

```diff
@@ -681,17 +681,17 @@ if [[ "$(zpool list)" == "no pools available" ]] \
         POOL_FILE=/mockcn/${MOCKCN_SERVER_UUID}/pool.json
     fi
 
-    dlargs=""
+    declare -a dlargs
 
     # The older config parameters lacked a 'disk_' prefix, but for the sake
     # sanity, we won't squat on CONFIG_exclude.
-    [[ "${CONFIG_cache}" == "false" ]] && dlargs+="-c"
-    [[ -n "${CONFIG_spares}" ]] && dlargs+="-s ${CONFIG_spares}"
-    [[ -n "${CONFIG_width}" ]] && dlargs+="-w ${CONFIG_width}"
-    [[ -n "${CONFIG_layout}" ]] && dlargs+="${CONFIG_layout}"
-    [[ -n "${CONFIG_disk_exclude}" ]] && dlargs+="-e ${CONFIG_disk_exclude}"
+    [[ "${CONFIG_cache}" == "false" ]] && dlargs+=("-c")
+    [[ -n "${CONFIG_spares}" ]] && dlargs+=("-s ${CONFIG_spares}")
+    [[ -n "${CONFIG_width}" ]] && dlargs+=("-w ${CONFIG_width}")
+    [[ -n "${CONFIG_disk_exclude}" ]] && dlargs+=("-e ${CONFIG_disk_exclude}")
+    [[ -n "${CONFIG_layout}" ]] && dlargs+=("${CONFIG_layout}")
 
-    if ! /usr/bin/disklayout ${dlargs} >${POOL_FILE}; then
+    if ! /usr/bin/disklayout ${dlargs[@]} >${POOL_FILE}; then
```

Comment:
And these should be quoted:

    if ! /usr/bin/disklayout "${dlargs[@]}" > "${POOL_FILE}"; then

Reply:
I don't believe so -- if it's quoted, `-e DISK DISK...` becomes a single argument -- as if you did `disklayout "-e DISK DISK..." ...` which is unlikely to do what is wanted here. I think we explicitly want it unquoted so that any embedded spaces are expanded out into individual arguments. Since the option arguments themselves are disks names which should never have embedded spaces, we shouldn't need to worry about that.

Reply:
Right, but that’s also why I said the args should be split above. If you do both they’ll be right.

Reply:
But then we'd need a way to split the `$CONFIG_xxx` values, which gets more complicated (since that has a lot more implications since the code that sets those gets used for everything else as well).

Reply:
Yeah, I suppose that’s true. Oh well.